lib/WUI/link_content/basic_gets.cpp

```diff
@@ -0,0 +1,223 @@
+#include "basic_gets.h"
+#include "filament.h" //get_selected_filament_name
+#include "../json_encode.h"
+#include "marlin_client.h"
+#include "lwip/init.h"
+#include "netdev.h"
+#include "../nhttp/segmented_json_macros.h"
+
+#include <cstring>
+#include <cstdio>
+
+namespace nhttp::link_content {
+
+JsonRenderer::ContentResult get_printer(size_t resume_point, JsonRenderer::Output &output) {
+    // Note about the marlin vars: It's true that on resumption we may get
+    // different values. But they would still be reasonably "sane". If we eg.
+    // finish a print and base what we include on previous version, we may
+    // outdated values, but they are still there.
+    marlin_vars_t *vars = marlin_vars();
+    const char *filament_material = get_selected_filament_name();
+
+    bool operational = true;
+    bool paused = false;
+    bool printing = false;
+    bool cancelling = false;
+    bool pausing = false;
+    bool ready = true;
+    bool busy = false;
+
+    marlin_client_loop();
+
+    switch (vars->print_state) {
+    case mpsPrinting:
+        printing = true;
+        ready = operational = false;
+        break;
+    case mpsPausing_Begin:
+    case mpsPausing_WaitIdle:
+    case mpsPausing_ParkHead:
+        printing = pausing = paused = busy = true;
+        ready = operational = false;
+        break;
+    case mpsPaused:
+        printing = paused = true;
+        ready = operational = false;
+        break;
+    case mpsResuming_Begin:
+    case mpsResuming_Reheating:
+    case mpsResuming_UnparkHead:
+        ready = operational = false;
+        busy = printing = true;
+        break;
+    case mpsAborting_Begin:
+    case mpsAborting_WaitIdle:
+    case mpsAborting_ParkHead:
+        cancelling = busy = true;
+        ready = operational = false;
+        break;
+    case mpsFinishing_WaitIdle:
+    case mpsFinishing_ParkHead:
+        busy = true;
+        ready = operational = false;
+        break;
+    case mpsAborted:
+    case mpsFinished:
+    case mpsIdle:
+    default:
```

Comment:
This should not be here because compiler won't check if all states are mentioned in the switch. And we want to handle all states including the future ones.

Reply:
Not a new code in this pull request, it got just moved from the original `wui_REST_api.c`; if you believe this should be solved, let's deal with it in a separate pull request.

Anyway, I think we have already talked about this and the issue is that it issues a warning _even if everything is handled_.